src/main/kotlin/ecommerce/model/Cart.kt

```diff
@@ -0,0 +1,71 @@
+package ecommerce.model
+
+import ecommerce.utils.exception.EntityNotFoundException
+import jakarta.persistence.CascadeType
+import jakarta.persistence.Entity
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.OneToMany
+import jakarta.persistence.OneToOne
+import kotlin.collections.minusAssign
+import kotlin.collections.plusAssign
+import kotlin.compareTo
+
+@Entity
+class Cart private constructor() {
```

Comment:
```suggestion
class Cart protected constructor() {
```

Sorry for the confusion 😅
We actually need a `public` or `protected` no-args constructor for JPA, not `private`.

Reply:
I was not able to do that but I added a library 
```
noArg {
    annotation("jakarta.persistence.Entity")
    annotation("jakarta.persistence.Embeddable")
    annotation("jakarta.persistence.MappedSuperclass")
}
```

i'll convert it to protected and then try

Reply:
But when I make it protected it says 
'protected' visibility is effectively 'private' in a final class 
And I have to add open as a prefix to class 
and if I add open it says
Redundant 'open' modifier 
How should I manage it

Reply:
`protected` constructors don't work as expected in Kotlin if a class is `final`, so it needs to be marked as `open`.
However, you can skip the `open` modifier when you use a JPA plugin like `allOpen` — it's added automatically 👀